tests/opentracing_instrumentation/test_requests.py

```diff
@@ -0,0 +1,53 @@
+import mock
+import pytest
+import requests
+
+
+from opentracing_instrumentation.client_hooks import requests as requests_hooks
+from opentracing_instrumentation.request_context import span_in_context
+
+
+@pytest.fixture
+def patch_requests(request):
+    if request.getfixturevalue('hook'):
+        response_handler_hook = mock.Mock()
+    else:
+        response_handler_hook = None
+
+    requests_hooks.install_patches(response_handler_hook)
+    try:
+        yield response_handler_hook
+    finally:
+        requests_hooks.reset_patches()
+
+
+@pytest.mark.parametrize('scheme', ('http', 'https'))
+@pytest.mark.parametrize('root_span', (True, False))
+@pytest.mark.parametrize('hook', (True, False))
+@mock.patch('requests.adapters.HTTPAdapter.cert_verify')
+@mock.patch('requests.adapters.HTTPAdapter.get_connection')
+def test_requests(get_connection_mock, cert_verify_mock,
```

Comment:
to confirm - these are in reverse order vs. annotations?

Reply:
Yes, `get_connection_mock` and `cert_verify_mock` are positional arguments which are passed into test function [from the bottom up](https://docs.python.org/3/library/unittest.mock.html#quick-guide) (sorry, there is no corresponding anchor at the documentation page).

> **Note:** When you nest patch decorators the mocks are passed in to the decorated function in the same order they applied (the normal Python order that decorators are applied). This means from the bottom up, so in the example above the mock for module.ClassName1 is passed in first.

```python
>>> import mock

>>> @mock.patch('requests.adapters.HTTPAdapter.cert_verify')
... @mock.patch('requests.adapters.HTTPAdapter.get_connection')
... def test_requests(get_connection_mock, cert_verify_mock):
...     print('get_connection_mock: {}'.format(get_connection_mock))
...     print('cert_verify_mock: {}'.format(cert_verify_mock))
...     

>>> test_requests()
get_connection_mock: <MagicMock name='get_connection' id='139800850314296'>
cert_verify_mock: <MagicMock name='cert_verify' id='139800850164928'>
```

Meanwhile, `scheme`, `root_span`, and `hook` can be reordered.
I didn't find an explanation in [corresponding documentation](https://docs.pytest.org/en/latest/parametrize.html), but I can give you a simple example.
```bash
$ cat test.py 
```
```python
import pytest


@pytest.mark.parametrize('a', ['a'])
@pytest.mark.parametrize('b', ['b'])
def test_ab(a, b):
    assert a == 'a'
    assert b == 'b'


@pytest.mark.parametrize('a', ['a'])
@pytest.mark.parametrize('b', ['b'])
def test_ba(b, a):
    assert a == 'a'
    assert b == 'b'
```
```bash
$ pytest test.py -v
======================== test session starts ========================
platform linux -- Python 3.6.7, pytest-4.0.1, py-1.7.0, pluggy-0.8.0 -- /home/mim/.local/share/virtualenvs/test-ghcGXTlL/bin/python3
cachedir: .pytest_cache
rootdir: /home/mim/projects/test, inifile:
collected 2 items                                                   

test.py::test_ab[b-a] PASSED                                  [ 50%]
test.py::test_ba[b-a] PASSED                                  [100%]

===================== 2 passed in 0.01 seconds ======================
```